UndertaleModLib/Scripting/IScriptInterface.cs

```diff
@@ -22,84 +24,469 @@ public ScriptException(string msg) : base(msg)
+        /// </summary>
+        /// <param name="status">The new status. For example <c>Decompiling...</c>.</param>
         void UpdateProgressStatus(string status);
+
+        //TODO: considering this forces everything that implements this to have their own progressValue,
+        //why not make that a necessary attribute?
+        /// <summary>
+        /// Adds a certain amount to the variable holding a progress value.
+        /// </summary>
+        /// <param name="amount">The amount to add.</param>
         void AddProgress(int amount);
-        void IncProgress();
-        void AddProgressP(int amount);
-        void IncProgressP();
+
+        /// <summary>
+        /// Increments the variable holding a progress value by one.
+        /// </summary>
+        void IncrementProgress();
+
+        /// <summary>
+        /// Adds a certain amount to the variable holding a progress value in.
+        /// Used for parallel operations, as it is thread-safe.
+        /// </summary>
+        /// <param name="amount">The amount to add.</param>
+        void AddProgressParallel(int amount);
+
+        /// <summary>
+        /// Increments the variable holding a progress value by one.
+        /// Used for parallel operations, as it is thread-safe.
+        /// </summary>
+        void IncrementProgressParallel();
+
+        /// <summary>
+        /// Gets the value of the variable holding a progress value.
+        /// </summary>
+        /// <returns>The value as <see cref="int"/>.</returns>
         int GetProgress();
+
+        /// <summary>
+        /// Sets the value of the variable holding a progress variable to another value.
+        /// </summary>
+        /// <param name="value">The new value for the progress variable.</param>
         void SetProgress(int value);
+
+        /// <summary>
+        /// Hides the progress bar.
+        /// </summary>
         void HideProgressBar();
+
+        /// <summary>
+        /// Enables the UI.
+        /// </summary>
         void EnableUI();
+
+        /// <summary>
+        /// Allows scripts to synchronize their assets with the lists from the UI.
+        /// </summary>
+        /// <param name="resourceType">A comma separated list of asset list names. This is case sensitive.</param>
+        /// <param name="enable">Whether to enable or disable the synchronization.</param>
+        //TODO: Having resourceType as a comma separated list just screams for error. Make it use some array of predefined assets it can use.
         void SyncBinding(string resourceType, bool enable);
-        void SyncBinding(bool enable = false);
-        void StartUpdater();
-        Task StopUpdater();
 
-        Task<bool> GenerateGMLCache(ThreadLocal<Decompiler.GlobalDecompileContext> decompileContext = null, object dialog = null, bool isSaving = false);
+        /// <summary>
+        /// Stops the synchronization of all previously enabled assets.
```

Comment:
_"Calls `BindingOperations.DisableCollectionSynchronization()` for all the asset lists that have enabled synchronization."_

Reply:
Disagree with this.
Different GUIs / implementations could handle this differently, there is no guarantee that this exact method will be called.

Reply:
There is no other method that does that _(at least for now)_.